### PR TITLE
MM-38774 - partial fix for: Community running out of memory

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- a fix to try to narrow down whether  generateMiniPreview is causing the community OOM issue
- add prefix `invalid-` to MimeType to prevent image from entering the generateMiniPreview

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-38774

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/file.go [hunk 1]
```diff
@@ -1129,6 +1129,15 @@ func (a *App) generateMiniPreview(fi *model.FileInfo) {
 			mlog.Debug("generateMiniPreview: prepareImage failed", mlog.Err(imgErr),
 				mlog.String("fileinfo_id", fi.Id), mlog.String("channel_id", fi.ChannelId),
 				mlog.String("creator_id", fi.CreatorId))
+
+			// Since this file is not a valid image (for whatever reason), prevent this fileInfo
+			// from entering generateMiniPreview in the future
+			fi.UpdateAt = model.GetMillis()
+			fi.MimeType = "invalid-" + fi.MimeType
+			if _, err2 := a.Srv().Store.FileInfo().Upsert(fi); err2 != nil {
```
Comment (cpoile) [inline app/file.go:1137]: err2 because err above is *model.AppError

File: app/file.go [hunk 2]
```diff
@@ -1129,6 +1129,15 @@ func (a *App) generateMiniPreview(fi *model.FileInfo) {
...
+				mlog.Debug("FileInfo save failed", mlog.Err(err2))
+			}
```
Comment (lieut-data) [inline app/file.go:1139]: nit: Propose avoiding `err2` by writing as (plus tweak to error text):
```suggestion
			err = a.Srv().Store.FileInfo().Upsert(fi); 
			if err != nil {
				mlog.Debug("Invalidating FileInfo failed", mlog.Err(err))
			}
```

**Review by lieut-data — APPROVED** [review]

File: app/file.go (hunk 1 above)
Comment (lieut-data) [reply app/file.go:1137]: Hah, I should have waited for this comment :p

**Review by jwilander — APPROVED** [review]

File: app/file.go (hunk 2 above)
Comment (cpoile) [reply app/file.go]: Fixed. :)

**Review by lieut-data — APPROVED** [review]

**Comment by jwilander** [comment]
@cpoile looks like some of the tests need updating

**Comment by agnivade** [comment]
Removed the AutoMerge label since we would be deploying from this branch directly. https://community-daily.mattermost.com/private-core/pl/837iriqphtfpdghp6rmtzkyw7c.

**Comment by agnivade** [comment]
@cpoile - FYI I pushed a new commit for debugging purposes into this branch so that we can deploy to daily.

**Review by jwilander — APPROVED** [review]

**Comment by amyblais** [comment]
@agnivade @cpoile Does this need to be cherry-picked?

**Comment by agnivade** [comment]
Nope, it's not important enough.

**Pull request merged**